### PR TITLE
.h splitting and reorg

### PR DESCRIPTION
**Pull request by lobradov**
- Isolated dependencies
- Sensors dependencies in individual sensor files.
- enums, macros and prototypes.

### CODE REVIEW

File: code/espurna/config/dependencies.h [hunk 1]
```diff
@@ -0,0 +1,76 @@
+#ifndef DEPENDENCIES_H
```
Comment (mcspr) [inline code/espurna/config/dependencies.h]: Just as sensors this should be pragma once.

Comment (lobradov) [reply code/espurna/config/dependencies.h]: good point, changed.

**Comment by xoseperez** [comment]
It's a good approach, but I'm worried about the dependencies. For instance: modifying ADC_VCC_ENABLED is useless since this setting is used to set de ADC_MODE at the bottom of the sensors.h file... Maybe that call (ADC_MODE) should be in the analog sensor classes, same thing you did for I2C.

**Comment by lobradov** [comment]
I see what you mean. I can do it - give me a minute.

**Comment by xoseperez** [comment]
Let's go!

**Comment by xoseperez** [comment]
I had to change some stuff (main problem was with DEBUG_MSG definition order but got it working. I think it's a good step. Things look more rational now.

**Pull request merged**